instrumentation/opentelemetry-instrumentation-pika/src/opentelemetry/instrumentation/pika/pika_instrumentor.py

```diff
@@ -126,12 +132,27 @@ def uninstrument_channel(channel: BlockingChannel) -> None:
             return
 
         for consumers_tag, client_info in channel._consumer_infos.items():
-            if hasattr(client_info.on_message_callback, "_original_callback"):
+            callback_attr = (
+                PikaInstrumentor._consumer_callback_attribute_name()
+            )
+            consumer_callback = getattr(client_info, callback_attr)
+            if hasattr(consumer_callback, "_original_callback"):
                 channel._consumer_infos[
                     consumers_tag
-                ] = client_info.on_message_callback._original_callback
+                ] = consumer_callback._original_callback
         PikaInstrumentor._uninstrument_channel_functions(channel)
 
+    @staticmethod
+    def _consumer_callback_attribute_name() -> str:
+        pika_version = version.parse(
+            pkg_resources.get_distribution("pika").version
+        )
+        return (
+            "on_message_callback"
+            if pika_version >= version.parse("1.0.0")
```

Comment:
Another thought I had: Since we can't have a mixture of versions, maybe we should just check this once when loading the library instead of when instrumenting and uninstrumenting each object.

Reply:
+1. Also doesn't pika have a version module that we can import and check instead of using `pkg_resources`? Ideally code should not rely on packaging systems to function.

Reply:
@owais It does, I'll use that instead. Thanks